packages/react/src/prop-types/deprecateValuesWithin.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright IBM Corp. 2016, 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { warning } from '../internal/warning';
+
+const didWarnAboutDeprecation = {};
+
+export default function deprecateValuesWithin(
+  propType,
+  message,
+  allowedValues
```

Comment:
I think this order of the params might be better? Also instead of a configurable message, what if we were able to pass in a mapping function that would be called (if available) to generate the message?

```suggestion
  propType,
  allowedValues
  propMappingFunction
```